src/fits.cc

```diff
@@ -848,10 +848,10 @@ void Fits::readTableArray(std::size_t row, int col, int nElements, T *value) {
     }
 }
 
-void Fits::readTableScalar(std::size_t row, int col, std::string &value) {
+void Fits::readTableScalar(std::size_t row, int col, std::string &value, bool isVariableLength) {
     int anynul = false;
-    long size = getTableArraySize(col);
-    // We can't directly write into a std::string (may be changing in C++11).
+    long size = isVariableLength ? getTableArraySize(row, col) : getTableArraySize(col);
```

Comment:
To expand on this: fits.h and fits.cc is low-level code and I'd rather keep it simple. Trying to do better in an efficient way would probably require adding some sort of cache to the `Fits` object for metadata about columns. I'm not even sure if such work is a good idea and it is more than I feel comfortable doing. So I propose to keep my solution.